src/main/kotlin/com/faforever/userservice/controller/OAuthController.kt

```diff
@@ -40,28 +42,31 @@ class OAuthController(
 ) {
     companion object {
         val LOG = LoggerFactory.getLogger(OAuthController::class.java)
+        const val LOGIN_TECHNICAL_ERROR_ROUTE = "login/technicalError"
     }
 
     @GetMapping("/login")
     fun showLogin(
         request: ServerHttpRequest,
+        response: ServerHttpResponse,
         @RequestParam("login_challenge") challenge: String,
         model: Model,
-    ): Mono<Rendering> {
-        val loginFailed = request.queryParams.containsKey("login_failed")
-        val loginThrottled = request.queryParams.containsKey("login_throttled")
-        model.addAttribute("loginFailed", loginFailed)
-        model.addAttribute("loginThrottled", loginThrottled)
-        model.addAttribute("challenge", challenge)
-        model.addAttribute("passwordResetUrl", fafProperties.passwordResetUrl)
-        model.addAttribute("registerAccountUrl", fafProperties.registerAccountUrl)
-        return Mono.just(Rendering.view("oauth2/login").build())
-    }
-
-    private fun redirect(response: ServerHttpResponse, uriString: String) = response.apply {
-        statusCode = HttpStatus.FOUND
-        headers.location = URI.create(uriString)
-    }.setComplete()
+    ): Mono<Rendering> =
+        hydraService.getLoginRequest(challenge)
+            .flatMap {
+                val loginFailed = request.queryParams.containsKey("login_failed")
+                val loginThrottled = request.queryParams.containsKey("login_throttled")
+                model.addAttribute("loginFailed", loginFailed)
+                model.addAttribute("loginThrottled", loginThrottled)
+                model.addAttribute("challenge", challenge)
```

Comment:
why not populate with the challenge returned by hydra here?

Reply:
But we don't need anything from this object.